### PR TITLE
Migrate to a recent version of asto + add a test for complex name

### LINKED ISSUE #4: Unable to upload by complex name
This adapter is working fine when I upload a file by repository name via Artipie:
I created repository config `simple.yaml`:
```yaml
repo:
  type: file
  storage:
    type: fs
    path: /tmp/artipie/repos/simple
```
and uploaded a file using:
```bash
http PUT http PUT http://localhost:8080/simple (some data)
```

But if I try to upload by file name:
```bash
http PUT http://localhost:8080/simple/filename (some data)
```

It's failing with error:
```
APP RxComputationThreadPool-3 com.artipie.vertx.DummySubscriber: Error occurred: io.vertx.core.file.FileSystemException: java.nio.file.FileSystemException: /tmp/artipie/repos/simple/simple/qwe: Not a directory
```

We need to be able to save files by complex keys/paths, e.g. upload `reponame/one/two/three/file` should work.

### PR DESCRIPTION
**Pull request by Sammers21**
Closes #4

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 I'm available and in need of reviews, please assign me

**Comment by 0crat** [comment]
This pull request [#5](https://github.com/artipie/files-adapter/issues/5) is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/7d155245-a917-406a-85dc-24b26c2c70c1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: pom.xml [hunk 1]
```diff
@@ -85,7 +85,7 @@ SOFTWARE.
     <dependency>
       <groupId>com.artipie</groupId>
       <artifactId>asto</artifactId>
-      <version>0.13.0</version>
+      <version>1.0-SNAPSHOT</version>
```
Comment (HDouss) [inline pom.xml]: @Sammers21 I think we should make a release of asto and use it here, we cannot rely on snapshot

File: src/test/java/com/artipie/files/FileSliceTest.java [hunk 2]
```diff
@@ -81,6 +81,37 @@ void putWorks() throws IOException {
...
     }
 
+    /**
+     * Put on complex name works correctly.
+     * @throws IOException if failed
+     */
+    @Test
+    void complexNamePutWorks() throws IOException {
+        final String hello = "Hello world!!!";
+        final Path temp = Files.createTempDirectory("temp");
```
Comment (HDouss) [inline src/test/java/com/artipie/files/FileSliceTest.java]: @Sammers21 We should delete this folder after the test. Better is to use TempDir annotation.

**Comment by HDouss** [comment]
@Sammers21 The build is failing

**Comment by Sammers21** [comment]
@HDouss , take another look, please.

**Comment by HDouss** [comment]
@g4s8 Approved. Please review and/or merge.

File: src/test/java/com/artipie/files/FileSliceTest.java [hunk 3]
```diff
@@ -81,18 +82,48 @@ void putWorks() throws IOException {
         vertx.close();
     }
 
+    /**
+     * Put on complex name works correctly.
+     * @param temp The temp dir.
+     * @throws IOException if failed
+     */
+    @Test
+    void complexNamePutWorks(@TempDir final Path temp) throws IOException {
```
Comment (g4s8) [inline src/test/java/com/artipie/files/FileSliceTest.java]: @Sammers21 let's move it to integration test, unit tests should not depend on external components and should pass in a few milliseconds

**Comment by Sammers21** [comment]
@g4s8 , done

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @hdouss[/z](https://www.0crat.com/u/hdouss)[/z](https://www.0crat.com/u/hdouss) didn't make CR comments

**Comment by 0crat** [comment]
Job was finished in 18 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**